cake-tech/cake_wallet
# WIP: CW-665 Implement AirGapped Monero Transactions

# Description

Work in progress for the airgap.

## How to test
1. Download offline wallet of choice
    1. xmruw - Unnamed Monero Wallet (project of mine, used during development)
       1. https://github.com/MrCyjaneK/unnamed_monero_wallet/releases/download/v1.0.0-RC61/app-clean_monero-release.apk 
       2. on _Let's start!_ screen click "Other options" (below "Normal Wallet" button) and select **Offline Wallet**
       3. you can either create a new wallet or restore existing (polyseed and legacy seeds are supported, with passphrase offsets)
       4. **optional** if you feel like breaking my work is a great idea you can go to 3 dot menu -> configuration -> and check showExtraOptions - it will show extra options in 3 dots menu that will allow you to generate URQRs that are not supported, cake wallet should handle all QR codes gracefully (i.e. throw error but not crash), also no matter what you scan it should work.
    2. anonero.io (tor browser required to visit link)
        1. download **ANON** flavor of the app, and on setup screen do not input node details, offline wallet will be created
    3. feather wallet
          1. (no idea, don't remember, please contibute)
2. Create view-only wallet
    1. Download cake wallet from this PR (last one with CW-665 in slack, or from github artifacts)
    2. Create a view-only wallet
        1. via keys -> input only primary address, secret spend key and restore height
        2. via qr code -> generate **NERO** format (supported in xmruw, anonero and I think feather)
3. Initial balance after sync may be wrong - open drawer on the right and use **Export outputs** to sync outputs to offline wallet, and get key images in exchange.
4. Create transaction in cake wallet, scan the codes using the QR code scanner icon located in right upper corner (in xmruw and anonero)


# Pull Request - Checklist  

- [ ] https://github.com/MrCyjaneK/fast_scanner/issues/20
- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods
